pkg/controller/import-controller_test.go

```diff
@@ -7,86 +7,45 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/util/diff"
-	kubeinformers "k8s.io/client-go/informers"
-	k8sfake "k8s.io/client-go/kubernetes/fake"
-	core "k8s.io/client-go/testing"
 	"k8s.io/client-go/tools/cache"
 
 	. "kubevirt.io/containerized-data-importer/pkg/common"
 )
 
-type importFixture struct {
-	t *testing.T
-
-	kubeclient *k8sfake.Clientset
-
-	// Objects to put in the store.
-	pvcLister []*corev1.PersistentVolumeClaim
-	podLister []*corev1.Pod
-
-	// Actions expected to happen on the client.
-	kubeactions []core.Action
-	actions     []core.Action
-
-	// Objects from here preloaded into NewSimpleFake.
-	kubeobjects []runtime.Object
+type ImportFixture struct {
+	ControllerFixture
 }
 
-func newImportFixture(t *testing.T) *importFixture {
-	f := &importFixture{}
-	f.t = t
-	f.kubeobjects = []runtime.Object{}
+func newImportFixture(t *testing.T) *ImportFixture {
+	f := &ImportFixture{
+		ControllerFixture: *newControllerFixture(t),
+	}
 	return f
 }
 
-func (f *importFixture) newController() *ImportController {
-
-	f.kubeclient = k8sfake.NewSimpleClientset(f.kubeobjects...)
-
-	podFactory := kubeinformers.NewSharedInformerFactory(f.kubeclient, noResyncPeriodFunc())
-	pvcFactory := kubeinformers.NewSharedInformerFactory(f.kubeclient, noResyncPeriodFunc())
-
-	podInformer := podFactory.Core().V1().Pods()
-	pvcInformer := pvcFactory.Core().V1().PersistentVolumeClaims()
-
-	c := NewImportController(f.kubeclient,
-		pvcInformer,
-		podInformer,
-		"test/myimage",
-		"Always",
-		"5")
-
-	for _, pod := range f.podLister {
-		c.podInformer.GetIndexer().Add(pod)
-	}
-
-	for _, pvc := range f.pvcLister {
-		c.pvcInformer.GetIndexer().Add(pvc)
+func (f *ImportFixture) newImportController() *ImportController {
+	return &ImportController{
+		Controller: *f.newController("test/myimage", "Always", "5"),
 	}
-
-	return c
 }
 
-func (f *importFixture) run(pvcName string) {
+func (f *ImportFixture) run(pvcName string) {
 	f.runController(pvcName, true, false, false)
 }
 
-func (f *importFixture) runWithExpectation(pvcName string) {
+func (f *ImportFixture) runWithExpectation(pvcName string) {
 	f.runController(pvcName, true, false, true)
 }
 
-func (f *importFixture) runExpectError(pvcName string) {
+func (f *ImportFixture) runExpectError(pvcName string) {
 	f.runController(pvcName, true, true, false)
 }
 
-func (f *importFixture) runController(pvcName string,
+func (f *ImportFixture) runController(pvcName string,
 	startInformers bool,
 	expectError bool,
 	withCreateExpectation bool) {
-	c := f.newController()
+	c := f.newImportController()
 	if startInformers {
 		stopCh := make(chan struct{})
 		defer close(stopCh)
```

Comment:
```go
cache.WaitForCacheSync(stopCh, c.pvcInformer.HasSynced)
cache.WaitForCacheSync(stopCh, c.pvcInformer.HasSynced)
```
Seems wrong as well, shouldn't we wait for Pvc AND Pod sync?

Reply:
Don't know. This is the original code. @davidvossel ?

Reply:
yeah, i don't know what's going on with that :/ It doesn't look right though.